inject-generator/src/main/java/io/avaje/inject/generator/TypeExtendsReader.java

```diff
@@ -119,15 +119,16 @@ String providesAspect() {
   }
 
   List<UType> autoProvides() {
-    if (!autoProvide || !providesAspect.isEmpty()) {
-      return List.of();
-    }
     if (baseTypeIsInterface) {
       return List.of(Util.unwrapProvider(baseUType));
     }
     var autoProvides = new ArrayList<>(interfaceTypes);
     autoProvides.addAll(extendsTypes);
-    autoProvides.add(Util.unwrapProvider(baseUType));
+    if (!autoProvide || !providesAspect.isEmpty()) {
+      autoProvides.remove(baseUType);
```

Comment:
Can you explain why this change was made?  Why was the early exit removed and changed to this?